### ISSUE TITLE
>Error: You forgot to solve the CAPTCHA. Please try again. >No valid captcha. 

### ISSUE THREAD

**Issue by anon0123** [ccd0/4chan-x#1300]
![20170220213734 4chan-x captcha not working](https://cloud.githubusercontent.com/assets/25911652/23142780/8c658688-f7b5-11e6-87be-88b429a23179.png)

>Error: You forgot to solve the CAPTCHA. Please try again. 
>No valid captcha. 

I can't post because these error messages just started coming up now. 

I'm using mozilla firefox portable esr 38.7.1 and 4chan x 1.13.8.3. 

Context (Don't have to read): Also. Was using firefox portable legacy esr 3.6.28, esr 10.0.11, esr 17.0.9, esr 24.7.0, and 28.0 before 4chan-x stopped working for those. So I decided to upgrade to 38 since some things weren't working and asked for that version specifically; And I was upgrading along the pattern of 18, 28, *8 in particularly anyway. 

I hope that the update fix for this is compatible with firefox 38. 

Context (Don't have to read): I mainly use portable programs; And wish all programs were portable. By far my largest folders are my www browser folders; Because I have multiple folders of them with different settings; Inb4 use profiles because those are saved locally and have trouble getting to work again when reformatting or if your browser gets hit by malware instead of "it just werks". I don't like firefox after 3 because it uses HTML5 which just seems to make everything bloated and spamvertising ad's far more difficult or even impossible to block; Just barely able to get by with multiple ad-blockers; Compared to firefox 3 and html4 where I just needed 1 of the most common adblocking program to have little to no issues. DRM shit. TBH. I also don't like how they're so much larger and bloated. Before this I didn't really mind any upgrades to firefox 3 before firefox started rapidly updating and ripping off of joogle chromium by the way. Firefox 18 was the last to have a "press esc to stop gif" feature that I liked. I was able to re-add it to 28. After 28 it became shit and a bad ripoff of joogle chromium. More so than before. I started using 38 after various things stopped working. I still have to use 28 for things like being able to access my router; Because 38 doesn't allow you to make an exception to some "untrusted connection" bullshit tbh. I also have to use 28 if I want to get rid of all the window's menu bar, navigation bar, and tab bar; Which 38 can't do. So having all these different versions tryharding to fix what isn't broken and being shitty DRM operating systems and being such a bitch to work instead of "just werks" is annoying. So I'd like some things to at least attempt to work with older versions. Or even at least have a "legacy" version that works in things like firefox 3 or some older versions. I recall there being some programs like that; That supported older or "legacy" versions. Sorry for the TL;DR context copy-pasta-like post that again you don't have to read; But so many people asked. 

Also (Don't have to read): I like posting anonymously without having to sign up for things. That's the point. I usually post about this on the image boards. Because it's from the image boards. So post about it there. Since it's not serious. Makes sense. But no. Everyone takes the internet seriously now. So everyone bitched in and told me not to post there. That's sad. So. Attempting to post on here. If all goes well then I will continue to post on here. If I get shit that reminds me of why I wanted to post anonymously on the image boards in the 1st place then I'm not doing this anymore. 

Also (Don't have to read): Added a picture on here. So hopefully it gets fixed faster today. As I don't have much to do other than stare at my computer screen until I can post again. Other suggestions on other things to do - when this apparently inevitably happens - would be appreciated, too, I guess... 

Also in the unfortunate event that you won't make it work for 38 for whatever reason then please recommend the minimal version needed; Most likely going to do 48 at this point for the sake of keeping a consistent *8 upgrade pattern. 

TL;DR: Fix 4chan-x. For firefox. Preferably working in firefox 38 or earlier if you can. 

Thank you.

**Comment by anon0123** [comment]
Also. 

>>>1300

Check 'em.

**Comment by ccd0** [comment]
Does the v2 version of the captcha work, or it it just v1?
Does it make a difference whether you're starting threads, or just replying to them?

**Comment by ccd0** [comment]
Sorry, I worded that badly:
Is the v2 version (images) of the captcha also broken, or it it just v1 (text)?

**Comment by anon0123** [comment]
I don't know. I don't use v2. 
I'm not sure because I can't start a thread to reply to it and I usually stick to my own threads.

**Comment by anon0123** [comment]
It's fine. I understood. I've been here so long. I know what things are. - I don't know. I don't use it. 

I'm gonna look into it more after this post. Thanks for the fast reply once again, based anon.

**Comment by ccd0** [comment]
What version of Greasemonkey are you using?

**Comment by ccd0** [comment]
It looks like 4chan has disabled starting new threads with the v1 Recaptcha. v2 should work.

They've done this before, but last time they changed it back shortly, so I don't know if this is temporary or permanent. You might want to bug https://www.4chan.org/feedback about restoring v1 support or, better, some other alternative captcha, especially if you don't like v2.

**Comment by anon0123** [comment]
>What version of Greasemonkey are you using?

Greasemonkey version 3.9

**Comment by anon0123** [comment]
>It looks like 4chan has disabled starting new threads with the v1 Recaptcha. v2 should work.
>
>They've done this before, but last time they changed it back shortly, so I don't know if this is temporary or permanent. You might want to bug https://www.4chan.org/feedback about restoring v1 support or, better, some other alternative captcha, especially if you don't like v2.

Seriously? 

Shoot. I'm shy and really bad at talking to people. I may make them do the opposite of what I say. So I'd appreciate it if you or someone else did it.

**Comment by anon0123** [comment]
I wonder why they're doing this. This is really stupid. I have to enable v2 just to start a thread? 

Absolutely ridiculous. 

Well. 

Thanks anyway.

**Comment by anon0123** [comment]
Alright. 

I made a file full of my firefox portable programs that have 4chan-x in it. I just cleared the history and removed the bookmarks. 

Because I figured it would be way easier. 

Now you can just see what it is for yourself. 

See. Portable programs are awesome right. I told you so!! I was right again!! - Enjoy!! 

https://www.mediafire.com/?5l2j77eebpqhbl7

Also: Again. Thanks for the sympathy. Even tho you didn't want to keep supporting 28. And do whatever the corporations tell us to do. I don't agree wiht it. But. "k den". So. 

I appreciate the fast replies and help. So. 

Thank you.

**Comment by anon0123** [comment]
Also. 


A /qa/ thread may be nice too. Or even worse. Considering it is apparently full of cancer that doesn't support chemo and actually fights against it. That's sad. 

Also if you or someone makes a feedback and it's public then post it so I can see that, too. 


Oh. And. Worse case scenario. Perhaps have a setting where you use the v2 captcha to make a new thread and the v1 captcha to make new posts. Still don't like, approve, or accept of mods doing this bullshit. Bitch is so bitch. But. Hey. Better than nothing. - I think that would be a good alternative scenario. Just for right now. 

Maybe even have that be a 3rd option. So we can turn it on by ourselves without bothering you. In case it is changed back and this happens again. If you want to. For right now. 

I probably won't post for some time until the mods stop this bullshit or you fix their bullshit. So if you could do that as fast as possible then it would be really appreciated. 


Thank you.

**Comment by ccd0** [comment]
You should look at what you can do with the v2 captcha with 4chan X. Most of the time you can fill it out with the keyboard using the `uiojklm,/` keys or the numpad. If you turn on "Force Noscript Captcha", you can avoid dealing with the obnoxious "click until there are none left" puzzles, which, at least to me, are more annoying than having to solve the captcha multiple times.

An option to use v1 for replies and v2 for starting threads would be ideal, but I'm worried about doing a bunch of work and then having v1 support dropped entirely. Recaptcha v1 has been abandoned by Google and isn't going to be available on 4chan for much longer. The question is how long it takes to go away. Could be today, could be next year.

**Comment by ccd0** [comment]
I've just pushed a quick workaround; now the "Use Recaptcha v1" option now only applies within threads, and there's a separate option (off by default) for the index and catalog. This isn't ideal yet because it doesn't let you use v1 when replying to a thread within the index; supporting that will require a bit more work (writing/testing code to switch between the captcha types). Might want to do #538 at the same time.

**Comment by anon0123** [comment]
>You should look at what you can do with the v2 captcha with 4chan X. Most of the time you can fill it out with the keyboard using the "uiojklm,/" keys or the numpad. If you turn on "Force Noscript Captcha", you can avoid dealing with the obnoxious "click until there are none left" puzzles, which, at least to me, are more annoying than having to solve the captcha multiple times.
>
>An option to use v1 for replies and v2 for starting threads would be ideal, but I'm worried about doing a bunch of work and then having v1 support dropped entirely. Recaptcha v1 has been abandoned by Google and isn't going to be available on 4chan for much longer. The question is how long it takes to go away. Could be today, could be next year.

Seriously? V2 captcha with "uiojklm,/" has a lot of issues. I started my own thread and it worked alright at 1st. Then when I got to the second captcha it stopped working. I also can't cache captchas with ctrl+shift; Which are extremely useful for image dumps; And moving quickly; Which was the entire reason image boards existed originally. Absolutely ridiculous. 

A better internet is worth fighting and dieing for. Somehow we came this far. I'm sure an update wouldn't be impossible. I mean we're here all the time. Why do we even bother doing this? Why would we waste time with captcha v2 for no reason? It would be great for right now. Maybe I would actually leave 4 if real captchas were gone entirely; As I'm even pushed towards doing now. Even if real captcha is said to have been abandoned by joogle - It's still up. And it still works. It not being available in the future remains to be seen. Maybe it will never go away. It could be never, could be never ever. There's no way to know for sure. Even if it does. I sometimes make an average of about 10 threads per day. Right now it's at about 0 to 1 threads per day since this incident. So you certainly will not be doing a bunch of work in vain. Same as always.

**Comment by anon0123** [comment]
My options. - I'm going to resist saying everything I want to say but just so you know what they're doing is bullshit. 

- >Constantly turn on or off captchas based on what I want to do. 
- >Have 2 separate versions of firefox. 1 for posting threads. 1 for making posts. And have to copy paste each individual thread to the "post" firefox. With all its bloat and precious resources it takes up. 

- >Don't make threads anymore. - Which is out of the question. Since I mainly do nothing -but- make and keep to my own threads most of the time most. 
- >tfw can't even have simplest thing ever

What I got. 

- >don't get the fix to having to use v2 captcha to make a thread
- >receive mods forced v2 captcha artificial problem that i never asked for

I never asked for this. 


So. How to solve the issues. 

How to solve the make-captcha-v2-somewhat-begrudgingly-acceptable problem. 

- >add ability to use the "tab" key to select something to activate captcha v2 - like when you use the "tab" key to bring up the captcha in captcha v1
- >get rid of the annoying "i am not a robot" button you have to click for every post
- >fix the issue of "uiojklm,/" not un-highlighting things sometimes
- >fix the issue of not being able to cache v2 captchas

Problem: captcha v2 is objectively inferior, slower, and more inefficient than captcha v1. 

How to solve not-being-able-to-use-captcha-v1-to-make-new-threads problem. 

- >mods enable captcha v1 to make new threads
- >the end

Problem: Mods. 
Solution: Mods listen to us and enable captcha v1 to make threads. 

How to solve the add-setting-to-use-captcha-v2-to-make-threads-and-captcha-v1-to-make-posts problem. 

- >add setting to use captcha v2 to make threads and captcha v1 to make posts
- >the end

Problem: The ones who are willing to do it are not able and the ones who are able to do it are not willing. Others are not willing or able. No one is willing and able. 

I just want to easily make a bunch of threads and posts without making "a thing" out of it. 

Talk normally and "it just works" vs. >hover to settings, >click settings, >scroll to "Use Recaptcha v1", >uncheck "Use Recaptcha v1", >hover to the "x", >"x" out of the settings window. 

Highlight captcha bar with tab and type captcha and post or cache and keep going vs. >hover to "captcha", >click "captcha", >hover to answer 1, >click answer 1, >hover to answer 2, >click answer 2, >hover to answer 3,  >click answer 3, >hover to answer 3, >hover to answer 4. 

Type captchas vs. >try to use "uiojklm,/", >screw up the 1st check because you weren't oriented properly, >completely unforgiving key-board shortcuts don't let you fix your mistake by clicking the same key again to uncheck them, >end up just usin the mouse anyway, >hover to answer 1, >click answer 1, >hover to answer 2, >click answer 2, >hover to answer 3,  >click answer 3, >hover to answer 3, >hover to answer 4. 

Why is this a thing? 

I suppose having a script to do this for me would be out of the question... 

What I want is so simple. That is the conflict. 

I'm already banned from and have to deal with bullshit everywhere else. And to be honest. That's the last thing I want to have to deal with here. 

Also. So the difference between posting on the image boards and github is that on the image boards I get to talk to more users about this and maybe get shit done and on github I get to talk to one person and don't get shit done. If everyone wants to know why I post on the image boards instead of github - That's why. - I feel divided and killed instead of multiplied and invincible. 

I may go back to image boards and keep making threads about this. Even if nothing gets done. Because now I can't do anything else. Now everything is shit. 

I resisted saying everything I want to say but just so you know what they're doing is bullshit.

**Comment by anon0123** [comment]
The very existence of 4chan-x on 4chan comes from the fact that mods force you to use their forced limited bullshit, leaving almost no place for choice. That's my problem. Mods/bans/cooldown/captcha/cancer/bullshit/etc every fucking day. And don't tell me "hurr this is how it is now". I still find original, funny shit where we make content for the lulz. It's just that it's suffocating under all the other cancer and bullshit. 4chan-x is not original, it's a reactionary move. 

We definitely have the power to rework everything if we keep at it. Like we used to before. Trying to make things better. I remember. 

We never thought we'd achieve shit, but at least we tried and had fun. Yeah. That's the internet I like. I remember.

**Comment by anon0123** [comment]
![](http://ac.pcriot.com/u/when%20someone%20won%27t%20help%20and%20you%20just.png)

**Comment by anon0123** [comment]
>> 	@ccd0 	Don't use v1 cached captchas for new threads. #1300

>> https://github.com/ccd0/4chan-x/commit/73c8054b6433a7c7255b8c56cb59d1b98d18c32a

>tfw

![](http://ac.pcriot.com/u/meme%20feels%20frog%20pepe%20crying%20smiling%20tears%20of%20joy.%201430102443651.jpg)

th...thanks

Thank you very much for all your hard work. 

![](http://ac.pcriot.com/u/-4.67.jpg)

Also I'd like it if we still went in the direction of making captcha v2 easier so it's more like captcha v1; With the issues I mentioned earlier; Son case if it ever is only captcha v2 with captcha v1 then we should at least try to make it easier like v1. But. Using cptcha v2 for threads and captcha v1 for posts is again a far easier option and I'm alright with this for now. 

Also "uiojklm,/"'s uncheck seems to work again. I'll hopefully assume it's fixed now. Thanks. 

"ctrl+shift" seems to work for captcha v2, too. I'll hopefully assume it's fixed now, too. Thanks. 

The main issue now is that a new captcha doesn't auto-appear in captcha v2; But it does auto-appear in captcha v1. This is really useful for image dumps or fast individual replies; Which I usually like doing. A trend I notice in the board-culture is that they actually get upset if reply to multiple users at hte same time. So I try to individualize it more. 

The 94x94 squares on a 9x9 grid of 282x282 pixels is quite large too. It'd be nice if there was an option to make them smaller to about 47x47 on a 9x9 grid of 141x141 pixels. Mainly because I work with multiple windows on a 1337x768 laptop and have my browser at a 341x384 resolution. I could increase this. And actually plan on upgrading to something with a 1920x1080 resolution which should solve the issue. But it'd still be nice to have the precious pixel space. For those that need it. I know joogle did it and the script. But if you can do something about it then do it. I think it as possible to resize captcha v1 to different sizes. 

Again. 

Captcha v1 is >tab, >type captcha, >cahce captcha, >type captcha, >cache captcha, >type captcha, >cahce captcha, >type captcha, >cache captcha. 

Captcha v2 is >hover, >click, >"uiojklm,/" capthca, >cache captcha, >hover, >click, >"uiojklm,/" capthca, >cache captcha. 

Captcha v2 should be >tab, >"uiojklm,/" capthca, >cahce captcha, >"uiojklm,/" capthca, >cache captcha, >"uiojklm,/" capthca, >cahce captcha, >"uiojklm,/" capthca, >cache captcha. 

An update fixing this wouldn't be for nothing. Since, as was brought up, it's possible captcha v2 will be the only option in the future. 

"Dig a well before you are thirsty". Be prepared. Just remember that and keep that in mind. But again this is fine for right now. 

![](http://ac.pcriot.com/u/vomitchan%20madness.%2004e60d48bd4d839e7062d1454714ef19.jpg)

Also. Since I've been accused of this in the past. I don't use this for bad purposes. I'm not bad... or at least not really bad... I actaully consider myself pretty nice... No. Killing someone for no reason is bad and evil. 

Again. I just like image dumping and just posting images and stuff without someone coming along and killing me for posting on the internet for no reason at all. I did it for the lulz. 

I am saddened by this forced "serious" posting and the cancer that supports it. Cancer mods and cancer users supporting cancer mods and cancer. Cancer. To be honest. Cancer. 

So. 

I'm sorry if I seem selfish and demanding.  It's just that it probably does help other users who don't bother asking. And some users actually do like me posting a lot. And most likely others, too. 

![](http://ac.pcriot.com/u/feel%20bad.%201366409902494.jpg)

I just want you to know. 

Your hard work is never for nothing. 

I appreciate the hard work you do. 

I was really stuck. 

You really help me out a lot. 

Thank you. 

![](http://ac.pcriot.com/u/vomit-chan%20sad%20happy%20smile.%20be5217cf7593e684bdd221bee69ce96c.jpg)

Jee... 

Thanks, ccd0... 

You're my best friend... 

...I love you... 

![](http://ac.pcriot.com/u/1334342443287.gif)

Thanks ccd0! 

I felt awful when I stopped by to post and now I feel better. 

Thank you!! 

![](http://ac.pcriot.com/u/hug.gif%201406688279606.gif%20va4%20ve1.gif)
